### ISSUE TITLE
Error message of `getindex(::Array, ::Nothing)` is unintuitive

### ISSUE THREAD

**Issue by Datseris** [JuliaLang/julia#32393]
Hi there, I am on 1.1.1. Here is an example case:
```
julia> x = rand(1000);

julia> x[nothing]
ERROR: ArgumentError: `nothing` should not be printed; use `show`, `repr`, or custom output instead.
Stacktrace:
 [1] print(::Base.GenericIOBuffer{Array{UInt8,1}}, ::Nothing) at .\show.jl:566
 [2] print_to_string(::String, ::Vararg{Any,N} where N) at .\strings\io.jl:123
 ...
```
This typically arises in cases where one uses `findfirst` and then tries to access `x` with the result of `findfirst`, which could be nothing.

Of course, I know that the result of `findfirst` should be handled first, but wouldn't it better if the result of `x[nothing]` showed a much better representation of what is happening, something like "You can't access array with `nothing`"?

**Comment by JeffBezanson** [comment]
Fixed in 1.2.

**Comment by JeffBezanson** [comment]
Dup of #30633. The fix could probably be backported though.
